reV/supply_curve/points.py

```diff
@@ -1469,6 +1469,67 @@ def points(self):
 
         return self._points
 
+    @staticmethod
+    def _chunk_excl(arr, resolution):
+        """Split an array into a list of arrays with len == resolution.
+
+        Parameters
+        ----------
+        arr : np.ndarray
+            1D array to be split into chunks.
+        resolution : int
+            Resolution of the chunks.
+
+        Returns
+        -------
+        chunks : list
+            List of arrays, each with length equal to self.resolution
+            (except for the last array in the list which is the remainder).
+        """
+
+        chunks = []
+        i = 0
+        while True:
+            if i == len(arr):
+                break
+            else:
+                chunks.append(arr[i:i + resolution])
+
+            i = np.min((len(arr), i + resolution))
+
+        return chunks
+
+    @staticmethod
+    def _excl_slices(arr, resolution):
+        """Split row or col ind into slices of excl rows or slices
+
+        Parameters
+        ----------
+        arr : np.ndarray
+            1D array to be split into slices
+        resolution : int
+            Resolution of the sc points
+
+        Returns
+        -------
+        slices : list
+            List of arr slices, each with length equal to self.resolution
+            (except for the last array in the list which is the remainder).
+        """
+
+        slices = []
+        i = 0
+        while True:
+            if i == len(arr):
+                break
+            else:
+                s, e = arr[i:i + resolution][[0, -1]]
+                slices.append(slice(s, e + 1))
+
+            i = np.min((len(arr), i + resolution))
```

Comment:
Same for this, ive recently been using a numpy array split then a list comprehension where `slices = [slice(chunk[0], 1 + chunk[-1]) for chunk in array_split]`

Reply:
what about this:
```
        slices = get_chunk_ranges(len(arr), resolution)
        slices = list(map(lambda i: slice(*i), slices))
 ```